lib/mayaUsd/fileio/primUpdaterManager.h

```diff
@@ -103,14 +153,64 @@ class PrimUpdaterManager : public PXR_NS::TfWeakBase
     //! Create the pull parent and set it into the prim updater context.
     MDagPath setupPullParent(const Ufe::Path& pulledPath, VtDictionary& args);
 
+    //! Record pull information for the pulled path, for inspection on
+    //! scene changes.
+    void recordPullVariantInfo(const Ufe::Path& pulledPath, const MDagPath& pullParentPath);
+
+    // Maya file new or open callback.  Member function to access other private
+    // member functions.
+    static void beforeNewOrOpenCallback(void* clientData);
+
+    void beginManagePulledPrims();
+    void endManagePulledPrims();
+
+    // Member function to access private nested classes.
+    static std::list<VariantSetDescriptor> variantSetDescriptors(const Ufe::Path& path);
+
     friend class TfSingleton<PrimUpdaterManager>;
 
+    class OrphanedNodesManager : public Ufe::Observer
```

Comment:
This is the object that manages pulled hierarchy visibility.  It observes the scene and looks for scene structure changes, upon which it will hide orphaned nodes, or show them when they are no longer orphaned.

Reply:
Does this class need to live within the other class? It might even be able to live entirely in an anoymous namespace in the cpp?

Reply:
(And if we have to add other thing to do when orphaned, it might even better live off as a separate class in its own header/cpp with a way to regsiter work to be done when a node is orphaned.)